.gitmodules

```diff
@@ -13,3 +13,9 @@
 [submodule "coap/libcoap"]
 	path = coap/libcoap
 	url = https://github.com/obgm/libcoap.git
+[submodule "json_generator/json_generator"]
+	path = json_generator/upstream
+	url = https://github.com/espressif/json_generator.git
+[submodule "json_parser/upstream"]
+	path = json_parser/upstream
+	url = https://github.com/espressif/json_parser.git
```

Comment:
Could we upload the components directly from these repositories? instead of creating wrappers here?

Reply:
The above repos are just mirrors of the original repos here, which are independent of esp-idf: https://github.com/shahpiyushv/json_parser, https://github.com/shahpiyushv/json_generator

Reply:
However the component URLs in idf_component.yml point to Espressif forks (e.g. https://github.com/espressif/json_generator).

Can we either redirect the submodule to Espressif fork or update the URLs?

Reply:
My bad, the submodule does point to Espressif fork, so everything looks to be in order.

The relationship between three repos is indeed a bit complicated, but let's keep it as proposed by @shahpiyushv for now. If it gets difficult to manage we can revisit this.